std/string.d

```diff
@@ -343,12 +343,38 @@ do
      return s;
      +/
 
-    // Need to make a copy
-    auto copy = new char[s.length + 1];
-    copy[0 .. s.length] = s[];
-    copy[s.length] = 0;
+    /+
+    Try use existent memory. Test reserved memory.
+    if reserved 
+      add trailing zero
+      return pointer
+    else
+      allocate new memory block
+      copy string
+      add zero
+      return pointer
+    +/
+
+    // reserve
+    if ( s.capacity > s.length )
+    {
+        char* cptr = cast( char* ) s.ptr; // C ptr
+        char* zptr = cptr + s.length;     // zero ptr
+        *zptr = '\0';        
 
-    return &assumeUnique(copy)[0];
+        return cast( immutable(char)* ) cptr;
```

Comment:
The same applies for this cast; no need for the white spaces. Also, you cannot return `cptr` since it is initialized with a `scope` variable (parameter `s`). That is why the autotester is complaining.